edx_analytics_transformers/processors/enterprise_processors.py

```diff
@@ -0,0 +1,51 @@
+"""
+Processor to add enterprise information for the user.
+"""
+from logging import getLogger
+
+from django.contrib.auth import get_user_model
+
+# pylint: disable=import-error
+from openedx.features.enterprise_support.api import EnterpriseApiClient, enterprise_enabled
+
+
+logger = getLogger(__name__)
+User = get_user_model()
+
+
+class EnterpriseContextProvider:
+    """
+    Processor to add enterprise information for the user.
+    """
+
+    def __call__(self, event):
+        """
+        Add enterprise information for the user in current
+        event
+        """
+        if not enterprise_enabled():
+            logger.info('Enterprise service is disabled. Cannot add enterprise information')
+            return event
+
+        username = event['context'].get('username')
+
+        if not username:
+            return event
+
+        try:
+            user = User.objects.get(username=username)
+            enterprise_learner_data = EnterpriseApiClient(user=user).fetch_enterprise_learner_data(user)
```

Comment:
@danialmalik 
This is going have performance implications. 
1. We need to check if enterprise app is enabled before this block of code
2. Can we introduce cache or some other way to avoid/minimize this call?

Reply:
@ziafazal 
1. Check is added at the start of `send` method.
2. Should we use `TieredCache`? We will have to cache the value for every user? Wouldn't this take up a lot in memory? What if we make store results for up to a number (say 1000) of users at a time in a python dict?